src/pages/LoginPage/LoginPage.tsx

```diff
@@ -0,0 +1,306 @@
+      img.src = FacebookIcon ?? '';
+    }
+  };
+
+  const handleSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+
+    const newErrors = { emailOrUsername: '', password: '' };
+
+    if (!emailOrUsername) {
+      newErrors.emailOrUsername = 'Email or Username is required.';
+    } else if (emailOrUsername.includes('@') && !EMAIL_REGEX.test(emailOrUsername)) {
+      newErrors.emailOrUsername = 'Please enter a valid email address.';
+    }
+
+    if (!password) {
+      newErrors.password = 'Password is required.';
+    } else if (password.length < PASSWORD_MIN_LENGTH) {
+      newErrors.password = `Password must be at least ${PASSWORD_MIN_LENGTH} characters long.`;
+    }
+
+    if (newErrors.emailOrUsername || newErrors.password) {
+      setErrors(newErrors);
+      return;
+    }
+
+    setErrors(newErrors);
+    onSubmit({ emailOrUsername, password });
+  };
+
+  const isDisabled = disabled || loading || !!errors.emailOrUsername || !!errors.password;
+
+  return (
+    <Box sx={getPageContainerStyles()}>
+      <Box component="form" onSubmit={handleSubmit} sx={getFormStyles(theme)}>
+        <Box sx={getLogoStyles(theme)}>
+          <Logo orientation="horizontal" width="100%" height="100%" alt="Logo" />
+        </Box>
+
+        <FormInput
+          label="Mail Id"
+          value={emailOrUsername}
+          onChange={(e) => {
+            setEmailOrUsername(e.target.value);
+            setErrors({ ...errors, emailOrUsername: '' });
+          }}
+          placeholder="Mail Id"
+          errorMessage={errors.emailOrUsername}
+          sx={getInputStyles(GEOMETRY_PX.EMAIL_USERNAME_INPUT_TOP, theme)}
+        />
+        {errors.emailOrUsername && (
+          <Typography
+            variant="caption"
+            sx={getErrorTextStyles(GEOMETRY_PX.EMAIL_USERNAME_ERROR_TOP, theme)}
+          >
+            {errors.emailOrUsername}
+          </Typography>
+        )}
+
+        <FormInput
+          label="Password"
+          type="password"
+          value={password}
+          onChange={(e) => {
+            setPassword(e.target.value);
+            setErrors({ ...errors, password: '' });
+          }}
+          placeholder="Password"
+          errorMessage={errors.password}
+          sx={getInputStyles(GEOMETRY_PX.PASSWORD_INPUT_TOP, theme)}
+        />
+        {errors.password && (
+          <Typography
+            variant="caption"
+            sx={getErrorTextStyles(GEOMETRY_PX.PASSWORD_ERROR_TOP, theme)}
+          >
+            {errors.password}
+          </Typography>
+        )}
+
+        <PrimaryButton
+          sx={{
+            ...getButtonStyles(theme),
+            // REFACTOR: Usar tokens del tema en lugar de colores harcodeados
+            backgroundColor: `${theme.palette.common.black} !important`,
+            '&:hover': {
+              backgroundColor: `${theme.palette.grey[900]} !important`,
+            },
+          }}
+          onClick={handleSubmit}
```

Comment:
The PrimaryButton is wired with onClick={handleSubmit} while the form has an onSubmit handler. Best practice is to set the button type to 'submit' and let the form call onSubmit, ensuring keyboard/assistive device triggers (Enter) work and semantics are preserved.

**Sugerencia:**
```typescript
Change the PrimaryButton usage to <PrimaryButton type="submit" loading={loading} disabled={isDisabled}>{buttonText}</PrimaryButton> and remove the onClick here. Keep the form's onSubmit={handleSubmit}.
```